docs/dcp2_system_design.rst

```diff
@@ -1223,49 +1246,105 @@ When a data release first enters the preparation phase, it is identical to the
+
+
+Alterations via ``complete`` staging areas
+------------------------------------------
+
+|nn| Before the ``delta`` staging areas specification was written, the DCP
+utilized the functionality that ``normal`` staging areas can contain multiple
+versions of an entity and reuse them to import the updates to a project. This
+is still being supported for backwards compatibility. |ne|
+
+
+Alterations via ``update`` staging areas
+----------------------------------------
+
+An ``update`` staging area facilitates the same alterations as a ``delta``
+staging area, albeit in a less space efficient form, and potentially causing
+redundant work in the importer, but it has the significant advantage of being
+easier to implement. This mechanism may take longer and maybe expensive for
+large datasets (e.g. Tabula Muris) in which case a ``delta`` staging area
+should be utilized. An ``update`` staging area has the ``type`` property set
+to ``udpate``. 
+
+An ``update`` staging area contains a complete set of subgraphs, metadata
+entities, and descriptors. The ``entity_id`` of entities, and the
+``links_id`` of subgraphs, must be stable. Only genuinely new entities
+(or subgraphs) may be assigned a new ``entity_id`` or ``links_id``. When the
+subgraph partitioning changes, the staging area source should minimize ID
+churn: if the metadata graph is partitioned into 100 subgraphs before an
+update, and 200 after, the source should write all 200 subgraphs with a new
+``links_id`` each. If instead the number of subgraphs remains the same but
+some or all of the subgraphs are subject to an update, like an added entity,
+the same ``links_id`` should be used for each updated subgraph, and any
+unchanged subgraphs should be written under their original ``links_id``,
+obviously. In between those two extremes (all new ``links_id`` values and all
+the same), the source is free to pick any suitable algortithm for allocating
+new ``links_id`` values and reusing existing ones.
+
+Before importing an ``update`` staging area, the importer empties all entity
+and links tables in the target dataset. The target dataset is the TDR dataset
+into which the staging area is being imported. It does not delete any data
+files. It then proceeds importing the ``update`` staging area as if it were a
+``complete`` staging area with the following exception: 
+
+If the importer encounters a descriptor that references a data file that is
+absent from the staging area, the importer checks for an existing data file
+in the target dataset. If one exists and has the CRC32C or SHA-256 checksum
+specified in the descriptor, the importer continues to import the descriptor,
+referencing the pre-existing data file in the target dataset. Note that an
+``update`` staging area may not alter the ``file_id``, ``file_version``,
+``size``, ``crc32c`` and ``sha256`` properties, it may alter or add other
+properties such as ``file_name`` or ``describedBy``. If there is no existing
```

Comment:
@hannes-ucsc  The `file_version` may be updated if the corresponding file metadata is updated.

Reply:
@ncalvanese1 if let's say the descriptor version was bumped up to a newer timestamp and the data file is not in the staging area, would this cause an issue on checking the data file existence in Terra?

Reply:
> the file metadata and descriptor file share the same uuid and version

The UUID in the descriptor path is the `entity.entity_id` as per spec it is not necessarily the same as the `descriptor.file_id`. Your source may conflate the two but I would not want to impose that on every source as it allows for having two metadata entities share a data file. Share in TDR, that is; sharing in a staging area is trivial even when `file_id` == `entity_id` but then the file would be duplicated on import.

Having `entity.version` == `descriptor.version` is also slightly problematic because it would mean that a mere metadata change incurs a descriptor change even though the rest of the descriptor content hasn't actually changed.

Assuming that ingest doesn't maintain a separate version for data files and that's likely not going to change anytime soon, I would be open to allowing `file_version` bumps.